src/transform/mod.rs

```diff
@@ -111,23 +112,23 @@ fn sinpi_arr(n: usize) -> &'static [i32; 5] {
 
 /// Utility function that returns the log of the ratio of the col and row sizes.
 #[inline]
-fn get_rect_tx_log_ratio(col: usize, row: usize) -> i8 {
+pub fn get_rect_tx_log_ratio(col: usize, row: usize) -> i8 {
   if col == row {
-    return 0;
+    return 0
```

Comment:
why remove the semicolon?

Reply:
I think I learned that rust syntax don't need it?
If you don't like it, let me restore them! :)

Reply:
I suggest keeping semicolons for explicit return statements, however I'm not sure that you need the `return` keyword at all in Rust, and we already use this shortcut in a few other places, so this might be better for consistency.

Reply:
For early returns, I believe you need the `return` keyword. If the return is part of the last expression in a function, you can omit the `return` and semicolon.

I'm a little surprised that this compiles successfully without the semicolon; from what I've seen, it seems to be the common practice to include the semicolon if you're using the `return` keyword.

Reply:
@lu_zero: Yes, I also think it needs 'return' to  exit the function there. Regarding whether ; is recommended and/or required, I'd ask lu_zero. In my understanding of how rustc parses, it can be omitted.

Reply:
Let me land this for now. I think that the reason why it can compile w/o ; is 1) there is only one statement, and the ending } is parsed as end of a complete statement, so not needing ;.
(Believe it or not).
If you ever feel not comfortable, please revise as you like as a separate patch. :)
I recall ',' is sometimes added at the end of last element of array, sometimes not.

Reply:
It works but it is a fun by-product, once I have a working keyboard I'll explain :)

The expression `return foo` has return type `!` (never).